doc/releases/changelog-dev.md

```diff
@@ -151,6 +151,13 @@
 
 The Operator class has undergone a major refactor with the following changes:
 
+* The static `compute_decomposition` method defines the decomposition
+  of an operator into a product of simpler operators, and the instance method
+  `decomposition()` computes this for a given instance. When a custom 
+  decomposition does not exist, the code now raises a custom `NoDecompositionError`
```

Comment:
I still like `DecompositonUndefined` more, because it states more what happened? Happy to be overruled though!